### PR TITLE
Update `wp-fork` to version `2.5.0`

### PR DESCRIPTION
**Pull request by geriux**
## Description

This PR updates the `wp-fork` branch with upstream version `2.5.0` and applies the required changes to publish the Android library associated with the package.

Since the PR contains a massive amount of changed files, in order to review this PR, it's better to compare the branch with the upstream repository and validate that we apply the same changes we have in `wp-fork`:
- [Changes applied on top of the `2.5.0` version of upstream repository](https://github.com/wordpress-mobile/react-native-gesture-handler/compare/upstream-2.5.0...wordpress-mobile:wp-fork-2.5.0)
- [Changes applied in `wp-fork`](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/2)

**NOTE:** Once this PR is ready, we should also merge https://github.com/wordpress-mobile/react-native-gesture-handler/pull/24 so the diff showed in `wp-fork` doesn't contain the `2.5.0` version changes.

### CODE REVIEW

**Comment by geriux** [comment]
**NOTE:** Before merging we need to [update this line](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/22/files#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65aR194) with the correct published Reanimated version.

**Review by derekblank — APPROVED** [review]
LGTM, pending updating the new published reanimated version. 👍 🚀

**Comment by geriux** [comment]
> LGTM, pending updating the new published reanimated version. 👍 🚀

Thanks for the reminder! Updated in https://github.com/wordpress-mobile/react-native-gesture-handler/pull/22/commits/3fafdab9e846bf8a469ea3924fefb4766dd11b65

**Pull request merged**
